LightAPRS-hab/LightAPRS-hab.ino

```diff
@@ -106,90 +171,201 @@ void setup() {
   APRS_useAlternateSymbolTable(alternateSymbolTable); 
   APRS_setSymbol(Symbol);
   //increase following value (for example to 500UL) if you experience packet loss/decode issues. 
-  APRS_setPreamble(350UL);  
+  APRS_setPreamble(500UL);  
   APRS_setPathSize(pathSize);
 
   configDra818(Frequency);
 
   bmp.begin();
  
 }
-
 void loop() {
   float tempC;
   float pressure;
-   wdt_reset();
-  
-  if (readBatt() > BattMin) {
-  
-  if(aliveStatus){
+  float loop_start = millis();
+
+  wdt_reset();
 
+  if (readBatt() > BattMin) {
+    if(aliveStatus) {
       //send status tx on startup once (before gps fix)
-      
       #if defined(DEVMODE)
         Serial.println(F("Sending"));
       #endif
+      
       sendStatus();
+      
       #if defined(DEVMODE)
         Serial.println(F("Sent"));
       #endif
-      
+
       aliveStatus = false;
-      
-   }
-    
-    updateGpsData(1000);
-    gpsDebug();
-
-    //debug for cjr
-    //sendStatus();
-    tempC = bmp.readTemperature();
-    pressure = bmp.readPressure();
-
-    if ((gps.location.age() < 1000 || gps.location.isUpdated()) && gps.location.isValid()) {
-      if (gps.satellites.isValid() && (gps.satellites.value() > 3)) {
-      updatePosition();
-      updateTelemetry();
-      
-      //GpsOFF;
-      setGPS_PowerSaveMode();
-      //GpsFirstFix=true;
+    }
 
-      if(autoPathSizeHighAlt && gps.altitude.feet()>3000){
+    if(secsTillTx <= 0) {
+      last_tx_millis = millis();
+      secsTillTx = GPSWait;
+
+      updateGpsData(1000);
+      gpsDebug();
+
+      if ((gps.location.age() < 1000 || gps.location.isUpdated()) && 
+          gps.location.isValid()) {
+        
+        if (gps.satellites.isValid() && 
+           gps.satellites.value() > 3) {
+
+          doAltCheck();
+          updateZone(); //updates BeaconWait
+          updateComment();
+          updatePosition();
+          updateTelemetry();
+          secsTillTx = BeaconWait;
+
+          //GpsOFF;
+          setGPS_PowerSaveMode();
+          //GpsFirstFix=true;
+          current_altitude = gps.altitude.feet();
+          if(current_altitude > max_altitude) {
+            max_altitude = current_altitude;
+          }
+          if(autoPathSizeHighAlt && gps.altitude.feet() > 3000){
             //force to use high altitude settings (WIDE2-n)
             APRS_setPathSize(1);
-        } else {
+          } else {
             //use defualt settings  
             APRS_setPathSize(pathSize);
-        }
-      
-      //send status message every 60 minutes
-      if(gps.time.minute() == 30){               
-        sendStatus();       
+          }
+          
+          //send status message every 15 minutes
+          if((gps.time.minute() % 15) == 0) {               
```

Comment:
in a follow on commit, let's see if we can make the status message sending a bit smarter, especially in the cases where we don't have GPS lock.  For example, to let us know how many satellites we have.

Reply:
Ok